ResidentContactApi.Tests/V1/Gateways/ResidentGatewayTests.cs

```diff
@@ -159,26 +163,112 @@ public void GetResidentByIdReturnsContactDetailsWithTypeAndSubtypeName()
         }
 
         [Test]
-        public void InsertedContactRecordShouldBeInsertedOnceInTheDatabase()
+        public void WhenGivenResidentIdInsertedContactRecordShouldBeInsertedOnceInTheDatabase()
         {
             var databaseEntity = AddPersonRecordToDatabase();
             var contactType = AddContactTypeToDatabase();
-            var contact = AddContactRecordToDatabase(databaseEntity.Id, contactType.Id);
 
-            var request = _fixture.Build<ResidentContact>()
+            var request = _fixture.Build<ContactDetailsDomain>()
                 .With(x => x.ResidentId, databaseEntity.Id)
                 .With(x => x.TypeId, contactType.Id)
                 .Without(x => x.SubtypeId)
                 .Create();
 
-            var response = _classUnderTest.InsertResidentContactDetails(request);
-            var databaseRecord = ResidentContactContext.Residents.Where(res => res.Id == response.Id);
+            var response = _classUnderTest.InsertResidentContactDetails(request.ResidentId, null, request);
+
+            var databaseRecords = ResidentContactContext.ContactDetails.Where(res => res.Id == response);
+            databaseRecords.Count().Should().Be(1);
+            var record = databaseRecords.First();
+
+            var expectedDatabaseRecord = new Contact
+            {
+                Id = response.Value,
+                ContactValue = request.ContactValue,
+                IsActive = request.IsActive,
+                IsDefault = request.IsDefault,
+                ContactTypeLookupId = contactType.Id,
+                ResidentId = databaseEntity.Id
+            };
+            record.Should().BeEquivalentTo(expectedDatabaseRecord, IgnoreForeignDatabaseObjects());
+        }
+
+        [Test]
+        public void WhenGivenAContactIdInsertContactDetailsShouldLinkToTheCorrectResident()
+        {
+            var person = AddPersonRecordToDatabase();
+            var contactId = AddCrmContactIdForResident(person);
+            var contactType = AddContactTypeToDatabase();
+
+            var request = _fixture.Build<ContactDetailsDomain>()
+                .With(x => x.ResidentId, person.Id)
+                .With(x => x.TypeId, contactType.Id)
+                .Without(x => x.SubtypeId)
+                .Create();
+
+            var response = _classUnderTest.InsertResidentContactDetails(null, contactId, request);
```

Comment:
Could we rename the response variable to have ID in the name e.g `responseId` since we're returning an ID from the method?